### ISSUE TITLE
Indicating whether posts have already been responded to

### ISSUE THREAD

**Issue by aaronpk** [indieweb/microsub#3]
The Microsub client needs to know whether a post has already been liked/replied/bookmarked/etc by the user in order to display a the appropriate interface buttons. Similarly the client should allow the user to delete the "like" post if it's been posted already, so it will need to know the URL of the like.

One way to accomplish this for external Microsub servers is the Microsub server can subscribe to your own site and index all the "like" posts it finds, and match those up. If your site is also your micropub and microsub server then all this is internal to the system and doesn't need to be specified.

**Comment by aaronpk** [comment]
I'm thinking about using a new Microsub property in the entry to indicate the response URLs that may exist for this post.

```
{
  "type": "entry",
  ...
  "_like": "https://aaronparecki.com/xxxxxxx",
  "_repost": "https://aaronparecki.com/yyyyyyyy"
}
``` 

alternately:

```
{
  "type": "entry",
  ...
  "_actions": {
    "like": "https://aaronparecki.com/xxxxxxx",
    "repost": "https://aaronparecki.com/yyyyyyyy"
  }
}
``` 

For likes and reposts, from a UI perspective, we want to know whether the user has already liked or reposted a post (a boolean yes/no), in order to style the like/repost button differently, so in that case it makes sense to have those be a single string. (The client could send a Micropub delete request with the URL if the user taps the button again, to delete the like/repost.)

For replies, I'm not sure what we actually want the client to indicate. Long-term it would be great if the client had the ability to show the full conversation thread, so in that sense it makes sense to handle replies differently from likes and reposts. Maybe we can handle this question separately for replies.

In the case of likes and reposts, we only care about the URLs of the user's own like/repost posts. Showing other people who have liked or reposted a post is a different problem, since that does not affect the state of the action buttons.

In thinking through this, I'm leaning towards the second JSON option above, nesting the data under `_actions`, since this data is limited in scope to help indicate the state of the action buttons. It is not meant to cover displaying all responses to a post.

**Comment by EdwardHinkle** [comment]
I agree, grouping it under _actions helps to logically group what like and repost mean, as opposed to having a _like and _repost entry-level property.

(Originally published at: https://eddiehinkle.com/2018/03/14/16/reply/)

**Comment by mblaney** [comment]
I've done this by extending webactions to return status urls, so it's all done client side. Saying that though, it puts extra burden on clients and requires individual sites to support returning the status urls, so getting the server to deal with it makes sense.

**Comment by pstuifzand** [comment]
A possibility could be to fetch feeds with likes, reposts, bookmarks. The server could match these up and return this as extra information. You could also see the likes and reposts of your friends and other people.
